### ISSUE TITLE
Getting error when trying to build book with newest version of bookdown

### ISSUE THREAD

**Issue by dchiu911** [rstudio/bookdown#1160]
When I update bookdown from 0.21 to 0.22, I am no longer able to build the book and get the following error:
```
Error in setwd(book_proj) : character argument expected
Calls: <Anonymous> -> site_generator -> create_site_generator -> setwd
Execution halted

Exited with status 1.
```
I also tried to dev version (0.22.2) and the error persists.

**Comment by yihui** [comment]
Could you provide the info required by the issue guide? https://github.com/rstudio/bookdown/issues/new?assignees=&labels=&template=bug-report.md&title= Without a reprex, it's hard to tell precisely what was wrong. My blind guess is that you don't have an index.Rmd that contains `site: bookdown::bookdown_site` in YAML.

**Comment by cderv** [comment]
Thanks for the report. We're are indeed missing some informations about how you end up with this error. Context about your usage, your project and what you did before encountering the error is really important us. Why ?

Usually, we have some tests that prevent us to introduce obvious issues, however we can't test all the usage that our user do with tools, and it is possible there is some edge cases we did not think of. That is why contextual information on what you did and what is you project is really important ! Only having a copy of the error message is not often enough to know what is going on. 

So here are some more hints based on the only information we have: 

I believe this error comes from here:
https://github.com/rstudio/bookdown/blob/d31bff356d51f815e6376cead02fc5cb998f9740/R/site.R#L14-L16

You must have clicked Render Book button to render the book in RStudio right ? 

It seems `book_proj` is not a character vector - this would mean that the book project path was not correctly found by `find_book_proj(input)` which did not returned a correct path. This should not happen if your project is a bookdown project containing a `index.Rmd` file with `site: bookdown::bookdown_site` in YAML. Hence the blind guess ☝️ . 

It would mean this would return nothing 
````r
bookdown:::find_book_proj(".")
````

So, 

* Is this a bookdown project ? 
* What is the structure of your project ? 
* How are you rendering the book ? 
* Is there anything special with your project organization , 
* Anything than can help us understand you specific use case ? 

Thanks for the additional information .

**Comment by dchiu911** [comment]
Hi @cderv thanks for the guidance on using `bookdown:::find_book_proj(".")`, I may have found the issue. It appears that this new helper function only permits an unquoted field for `site`, i.e.
```yaml
site: bookdown::bookdown_site
```
works but
```yaml
site: "bookdown::bookdown_site"
```
no longer does after the update (v0.21 doesn't introduce this condition yet). Most of my projects actually use the unquoted site but I happened to quote it for a new project (copied index.Rmd over from an old project) and stumbled on this issue. Perhaps nothing to fix here but maybe it would be best to update the bookdown usage page as it uses the quoted site?

https://bookdown.org/yihui/bookdown/usage.html

---
Also for future inquiries, I will add the session info (I optimistically thought it would be sufficient to simply note the versions here), but in terms of a reprex, what would be the best way to create this with a bookdown project private repo? Should I copy over the `index.Rmd`, `_bookdown.yml`, `_output.yml` and all the source R markdown files needed to generate the report with deidentified/anonymized text?

**Comment by yihui** [comment]
Usually we need a reprex so we can identify the bug by ourselves, since users may be less familiar with our codebase. In this case, since you have figured out the problem (i.e. you did the work that we would have to do), that's enough for us to fix the bug. Thanks!

**Comment by Cameron-Fairfield** [comment]
I faced a very similar bug in Bookdown projects. Even when creating a new Bookdown project with just the basic template I would get an error when clicking "Build Book" or running: `bookdown::render_book("index.Rmd", "bookdown::pdf_book")`

The error was: `Error in dirname(input) : a character vector argument expected`

Reverting to an older version of `Rmarkdown` seemed to fix it (v2.6 worked but v2.8 did not) but our group has just updated to 2.8.

Thanks @yihui. The latest `Bookdown` GitHub version has fixed the issue. It now compiles bookdown projects again even with `Rmarkdown` version 2.8.
